test/WebSites/BasicWebSite/Controllers/HomeController.cs

```diff
@@ -16,7 +16,9 @@ public IActionResult Index()
             return View();
         }
 
-        public IActionResult PlainView()
+        // Keep the return type as object to ensure that we don't
+        // wrap IActionResult instances into ObjectResults.
```

Comment:
I was thinking more along the lines of
> This is a functional test covering the case where a method returns an `IActionResult` but has a declared return type of `object. The invoker has to treat this as a special case, do not simplify this code.

This says "what's important" and "why it's important"